docs/cody/model-configuration/index.mdx

```diff
@@ -0,0 +1,553 @@
+field), custom models can be specified for that provider by adding them to the `"modelConfiguration.modelOverrides"` section.
+
+This field is an array of items, each with the following fields:
+
+-   `modelRef` - Uniquely identifies the model within the provider namespace.
+    -   A string in the format `${providerId}::${apiVersionId}::${modelId}`.
+    -   To associate a model with your provider, `${providerId}` must match the provider’s ID.
+    -   `${modelId}` can be any URL-safe string.
+    -   `${apiVersionId}` specifies the API version, which helps detect compatibility issues between models and Sourcegraph instances.
+        For example, `"2023-06-01"` can indicate that the model uses that version of the Anthropic API. If unsure, you may set this to `"unknown"` when defining custom models.
+-   `displayName` - An optional, user-friendly name for the model. If not set, clients should display the `ModelID` part of the `modelRef` instead (not the `modelName`).
+-   `modelName` - A unique identifier used by the API provider to specify which model is being invoked.
+    This is the identifier that the LLM provider recognizes to determine the model you are calling.
+-   `capabilities` - A list of capabilities that the model supports. Supported values: "autocomplete" and "chat".
+-   `category` - Specifies the model's category, with the following options:
+
+    -   `"balanced"` - Typically the best default choice for most users. This category is suited for state-of-the-art models, like Sonnet 3.5 (as of October 2024).
+    -   `"speed"` - Ideal for low-parameter models that may not be suited for general-purpose chat but are beneficial for specialized tasks, such as query rewriting.
+    -   `"accuracy"` - Reserved for models, like OpenAI o1, that use advanced reasoning techniques to improve response accuracy, though with slower latency.
+    -   `"other"` - Used for older models without distinct advantages in reasoning or speed. Select this category if uncertain about which category to choose.
+    -   `"deprecated"` - For models that are no longer supported by the provider and are filtered out on the client side (not available for use).
+
+-   `contextWindow` - An object that defines the number of "tokens" (units of text) that can be sent to the LLM.
+    This setting influences response time and request cost, and may vary according to the limits set by each LLM model or provider.
+    It includes two fields:
+    -   `maxInputTokens` - Specifies the maximum number of tokens for the contextual data in the prompt (e.g., question, relevant snippets).
+    -   `maxOutputTokens` - Specifies the maximum number of tokens allowed in the response.
+-   `serverSideConfig` - Additional configuration for the model. Can be one of the following:
+
+    -   `awsBedrockProvisionedThroughput` - Specifies provisioned throughput settings for AWS Bedrock models, with the following fields:
```

Comment:
@slimsag, do we have a deployment available to test this setup, or should we note that it’s a special case and recommend contacting Sourcegraph support for proper configuration?

Reply:
Unfortunately I'm not sure, we might need to ask @aramaraju or @kalanchan if they've seen an AWS bedrock testing environment floating around somewhere. Otherwise what you suggested sounds reasonable to me.

I know we have ~2 customers using this though.

Reply:
yes, will message you on slack!